src/main/java/gregtech/common/tileentities/machines/multi/MTEEnvironmentallyControlledChemicalFacility.java

```diff
@@ -0,0 +1,1308 @@
+                .dot(2)
+                .buildAndChain(GregTechAPI.sBlockCasings8, 0))
+        .addElement('F', ofFrame(Materials.Polytetrafluoroethylene))
+        .addElement('G', ofBlockAnyMeta(sBlockTintedGlass, 1))
+        .build();
+
+    public MTEEnvironmentallyControlledChemicalFacility(final int aID, final String aName, final String aNameRegional) {
+        super(aID, aName, aNameRegional);
+    }
+
+    public MTEEnvironmentallyControlledChemicalFacility(String aName) {
+        super(aName);
+    }
+
+    @Override
+    public IStructureDefinition<MTEEnvironmentallyControlledChemicalFacility> getStructureDefinition() {
+        return STRUCTURE_DEFINITION;
+    }
+
+    @Override
+    public boolean isCorrectMachinePart(ItemStack aStack) {
+        return true;
+    }
+
+    @Override
+    public IMetaTileEntity newMetaEntity(IGregTechTileEntity aTileEntity) {
+        return new MTEEnvironmentallyControlledChemicalFacility(this.mName);
+    }
+
+    @Override
+    public ITexture[] getTexture(IGregTechTileEntity aBaseMetaTileEntity, ForgeDirection side, ForgeDirection aFacing,
+        int colorIndex, boolean aActive, boolean redstoneLevel) {
+        if (side == aFacing) {
+            if (aActive) return new ITexture[] { casingTexturePages[1][48], TextureFactory.builder()
+                .addIcon(OVERLAY_FRONT_LARGE_CHEMICAL_REACTOR_ACTIVE)
+                .extFacing()
+                .build(),
+                TextureFactory.builder()
+                    .addIcon(OVERLAY_FRONT_LARGE_CHEMICAL_REACTOR_ACTIVE_GLOW)
+                    .extFacing()
+                    .glow()
+                    .build() };
+            return new ITexture[] { casingTexturePages[1][48], TextureFactory.builder()
+                .addIcon(OVERLAY_FRONT_LARGE_CHEMICAL_REACTOR)
+                .extFacing()
+                .build(),
+                TextureFactory.builder()
+                    .addIcon(OVERLAY_FRONT_LARGE_CHEMICAL_REACTOR_GLOW)
+                    .extFacing()
+                    .glow()
+                    .build() };
+        }
+        return new ITexture[] { casingTexturePages[1][48] };
+    }
+
+    @Override
+    protected MultiblockTooltipBuilder createTooltip() {
+        MultiblockTooltipBuilder tt = new MultiblockTooltipBuilder();
+        tt.addMachineType("Environmentally Controlled Chemical Facility, ECCF")
+            .addInfo(
+                EnumChatFormatting.GRAY + "Allows to produce some chemicals in a "
+                    + EnumChatFormatting.BLUE
+                    + "single step"
+                    + EnumChatFormatting.GRAY
+                    + ", but requires special conditions")
+            .addInfo(
+                EnumChatFormatting.GRAY + "Doesn't overclock, instead increases parallels by "
+                    + EnumChatFormatting.GOLD
+                    + "4 ^ (Energy Hatch Tier - Recipe EU tier) * parallel module values")
```

Comment:
```suggestion
                    + "4 ^ (Energy Hatch Tier - Recipe tier) * Parallel Modules")
```
Then add a line stating what Parallel Modules is (the _product_ of the "value" (tier function?) of both parallel modules)